opal/corelib/string.rb

```diff
@@ -1696,23 +1696,113 @@ def self._load(*args)
   def unpack(pattern)
     %x{
       function stringToBytes(string) {
-        var i,
-            singleByte,
-            l = string.length,
-            result = [];
+        var result = [];
 
-        for (i = 0; i < l; i++) {
-          singleByte = string.charCodeAt(i);
-          result.push(singleByte);
+        for (var i = 0; i < string.length; i ++) {
+          var chr = string.charCodeAt(i);
+          var byteArray = [];
+
+          while (chr > 0) {
+            byteArray.push(chr & 0xFF);
+            chr >>= 8;
+          }
+
+          // all utf-16 characters are two bytes
+          if (byteArray.length == 1) {
+            byteArray.push(0);
+          }
+
+          // assume big-endian
+          result = result.concat(byteArray.reverse());
         }
+
         return result;
       }
+
+      function stringToCodePoints(string) {
+        var result = [];
+
+        for (var i = 0; i < string.length; i ++) {
+          codePoint = charCodeAt(string, i);
```

Comment:
`codePoint` is a global variable here because it is not defined with `var` in this function (see https://travis-ci.org/opal/opal/jobs/157245342#L298)